ch_tools/chadmin/cli/object_storage_group.py

```diff
@@ -172,7 +174,17 @@ def _clean_object_storage(
     Delete orphaned objects from object storage.
     """
     disk_conf: S3DiskConfiguration = ctx.obj["disk_configuration"]
-    prefix = object_name_prefix or disk_conf.prefix
+
+    if object_name_prefix is not None:
+        prefix = object_name_prefix
+    else:
+        prefix = disk_conf.prefix
+        # Yandex cloud specific
+        # Remove the last shard section from prefix
+        if on_cluster:
+            match = re.match(r"(.*)shard\d+/$", prefix)
```

Comment:
Shard name can be an arbitrary string and can mismatch `shard\d+`.

Suggest to replace the regexp with:
```
prefix = os.path.dirname(prefix)
```

And I guess it's not Yandex Cloud specific. We just imply that shard paths are on the same nesting level.